### PR TITLE
docs: add logo and mascot

### PR DESCRIPTION
**Pull request by somethingelseentirely**
This adds a logo and mascot under the `branding` directory.

Merging is blocked until the [Python Software Foundation has given their OK](https://github.com/PyO3/pyo3/issues/3109#issuecomment-2323339368) for use of the python logos' color pattern.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
I really like the art!

File: branding/pyo3logo.ai
Comment (davidhewitt) [inline branding/pyo3logo.ai:1]: If it's ok by you, I might create a `PyO3/branding` repository and move this file there? I think same for the `.pixaki` files too.

**Review by davidhewitt — APPROVED** [review]
@somethingelseentirely, thanks again for this. The PSF has said that this is ok (quote: "inspired by [the Python logo], not a derivative"). If you are happy with it, can I merge this week and use it in the 0.23 release?

Sorry for the long delay (entirely on my part, the PSF was actually quite fast to get back to me, many thanks @malemburg).

File: branding/pyo3logo.ai
Comment (somethingelseentirely) [reply branding/pyo3logo.ai:1]: Yeah makes total sense! No need to burden people with bigger repo size. 😄
And really no worries, we're all doing this in our spare time 😆 it takes as long as it takes! ✨

**Comment by davidhewitt** [comment]
Superseded in #4697

**Pull request closed without merging**